File: 백준/Silver/2167. 2차원 배열의 합/2차원 배열의 합.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())

arr = [list(map(int, input().split())) for _ in range(N)]

for _ in range(K := int(input())):
    partialSum = 0
    x1, y1, x2, y2 = map(int, input().split())
    for i in range(x1, x2 + 1):
        for j in range(y1, y2 + 1):
            partialSum += arr[i-1][j-1]
    print(partialSum)

File: 백준/Silver/2167. 2차원 배열의 합/2차원 배열의 합.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

let line = 0;
const [N, M] = input[line++].split(' ').map(Number);

const arr = [];
for (let i = 0; i < N; i++) {
    arr.push(input[line++].split(' ').map(Number));
}

const K = Number(input[line++]);
for (let i = 0; i < K; i++) {
    let partialSum = 0;
    const [x1, y1, x2, y2] = input[line++].split(' ').map(Number);
    
    for (let r = x1 - 1; r <= x2 - 1; r++) {
        for (let c = y1 - 1; c <= y2 - 1; c++) {
            partialSum += arr[r][c];
        }
    }
    
    console.log(partialSum);
}